lib/highwaterService.js

```diff
@@ -135,13 +133,31 @@ module.exports = function highwaterService(userApiClient, envConfig, eventsProdu
     }, envConfig.httpsConfig), highwaterApi, userApiClient);
   }
 
+  function beforeShutdown() {
+    // avoid running into any race conditions
+    // https://github.com/godaddy/terminus#how-to-set-terminus-up-with-kubernetes
+    return new Promise(resolve => setTimeout(resolve, 5000));
+  }
+
+  async function onShutdown() {
+    log.info('Stopping the Highwater API server');
+    server.close();
+    log.info('Stopping the Kafka producer');
+    await eventsProducer.Stop()
+    return
+  }
+
   return {
-    close: function () {
-      log.info('Stopping the Highwater API server');
-      server.close(eventsProducer.stop);
-    },
     start: function (cb) {
       log.info('Start Highwater API server serving on port[%s]', servicePort);
+      createTerminus(server.server, {
+        healthChecks: {
+          '/status': highwaterApi.status,
```

Comment:
The status function should return a promise according to https://github.com/godaddy/terminus#usage. You can just declare it async.